sourced/ml/transformers/basic.py

```diff
@@ -153,13 +166,19 @@ def __setstate__(self, state):
         from bblfsh import Node
         self.parse_uast = Node.FromString
 
-    def __call__(self, rows: RDD) -> RDD:
+    def __call__(self, rows: Union[RDD, DataFrame]) -> RDD:
+        if isinstance(rows, DataFrame):
+            return rows.rdd.flatMap(self.deserialize_uast)
         return rows.flatMap(self.deserialize_uast)
 
     def deserialize_uast(self, row: Row):
         if not row[EngineConstants.Columns.Uast]:
             return
         row_dict = row.asDict()
-        row_dict[EngineConstants.Columns.Uast] = self.parse_uast(
-            row[EngineConstants.Columns.Uast][0])
+        try:
+            row_dict[EngineConstants.Columns.Uast] = self.parse_uast(
+                row[EngineConstants.Columns.Uast][0])
+        except Exception as e:
```

Comment:
The exception during parsing a UAST is impossible. If it happens, we need to fix the problem itself.